### PR TITLE
Add tests for dup attributes

### PR DESCRIPTION
**Pull request by mximp**
Related to #1807.

Adding two XSL tests proving current (broken) logic of cooperative execution of `abstracts-float-up.xsl` and `fix-missed-names.xsl`.

---

## PR-Codex overview
This PR adds a fix to the `eo-parser` module to handle missed names and includes tests to ensure correct behavior. 

### Detailed summary
- Added `fix-missed-names.xsl` to `eo-parser/src/main/resources/org/eolang/parser/optimize/`
- Added tests to `eo-runtime/src/test/eo/org/eolang/runtime-tests.eo`
- Added tests to `eo-parser/src/test/resources/org/eolang/parser/packs/float-up-same-attrs.yaml`
- Added tests to `eo-parser/src/test/resources/org/eolang/parser/packs/optimize/fix-names-with-duplicates.yaml`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Comment by mximp** [comment]
/codex What changes are made within existing tests?

**Comment by mximp** [comment]
@volodya-lombrozo @Graur, please check the changes

**Review by volodya-lombrozo — CHANGES_REQUESTED** [review]
@mximp As I understood, you added the tests that shows the wrong behaviour of parsing chain (not unique names), please, correct me if I'm wrong.
Would it be better to write tests that will expect correct behaviour instead of emphasising the wrong behaviour? In other words, I would write the test that clearly shows our expectations from the future changes and disable it by using `skip:true` option.

**Comment by mximp** [comment]
> @mximp As I understood, you added the tests that shows the wrong behaviour of parsing chain (not unique names), please, correct me if I'm wrong. Would it be better to write tests that will expect correct behaviour instead of emphasising the wrong behaviour? In other words, I would write the test that clearly shows our expectations from the future changes and disable it by using `skip:true` option.

@volodya-lombrozo The problem is that we don't know the expected behaviour. These two tests are exactly what happens now, so that the logic is clear. 
Next step would be describe new behaviour and fix the code (along with tests).

**Review by volodya-lombrozo — APPROVED** [review]
@mximp Looks good to me

**Comment by yegor256** [comment]
@mximp I also think that it's not a good idea to add such files to the repository, since we may simply forget in the future that they are actually emphasising the problem, instead of confirming the correct behavior. Maybe we can think about the "desired" behavior and express it in the tests?

**Comment by yegor256** [comment]
@mximp indeed, small commits/PRs are great, but each of them must move the system from one consistent state to another consistent state. In other words, we must always assume that after your commit you can leave the project and the knowledge of your intent will be lost. New developers will open the code base, see your changes, and will assume that this is a consistent state. They will not know that you actually wanted to remove these tests in the future, as invalid.

> And if I add tests with desired behaviour I will have to supply supporting functionality or disable it and create puzzle. I see no point in doing this.

Why no point? This is exactly what would be great in this case: disable the tests and create the puzzle.

**Comment by mximp** [comment]
@yegor256 @volodya-lombrozo please check now. I've changed base test to reflect expected behaviour. The second test left as-is as it probably correct.

File: eo-parser/src/test/resources/org/eolang/parser/packs/float-up-same-attrs.yaml [hunk 1]
```diff
@@ -0,0 +1,31 @@
+xsls:
+  - /org/eolang/parser/optimize/remove-refs.xsl
+  - /org/eolang/parser/optimize/abstracts-float-up.xsl
+  - /org/eolang/parser/optimize/remove-levels.xsl
+  - /org/eolang/parser/add-refs.xsl
+tests:
+  - /program[count(.//o[@base='build' and not(@ref)])=2]
+  - /program/objects[count(o[@original-name='build'])=2]
```
Comment (volodya-lombrozo) [inline eo-parser/src/test/resources/org/eolang/parser/packs/float-up-same-attrs.yaml:8]: @mximp You can add expectations about new name that both objects will have

**Review by volodya-lombrozo — CHANGES_REQUESTED** [review]

File: eo-parser/src/test/resources/org/eolang/parser/packs/float-up-same-attrs.yaml (hunk 1 above)
Comment (mximp) [reply eo-parser/src/test/resources/org/eolang/parser/packs/float-up-same-attrs.yaml:8]: @volodya-lombrozo I believe it doesn't matter for this test case: we don't care about the names here.

Comment (mximp) [reply eo-parser/src/test/resources/org/eolang/parser/packs/float-up-same-attrs.yaml:8]: @volodya-lombrozo It tests particular fact about the output xml. The fact is "we create two objects with the same base and two objects with same original name". For this test it's enough. There is no point in going deeper (which names objects have or what order they are placed in etc.). Just minimum enough to reveal the issue under consideration.

**Comment by mximp** [comment]
@volodya-lombrozo @yegor256 
I've added the fix itself. Please check.

**Comment by mximp** [comment]
@yegor256 please merge

**Comment by yegor256** [comment]
@rultor merge

**Pull request merged**